packages/test/stochastic-test-utils/src/random.ts

```diff
@@ -2,24 +2,80 @@
+export function makeUuid4(u32_0: number, u32_1: number, u32_2: number, u32_3: number) {
+    const hex = (value: number, digits: number) => value.toString(16).padStart(digits, "0");
+
+    return `${
+        hex(u32_0, 8)
+    }-${
+        hex(u32_1 >>> 16, 4)
+    }-${
+        // Discard low 4 bits and insert version '0b100'.
+        hex(((u32_1 << 16) >>> 20) | 0x4000, 4)
+    }-${
+        // Insert variant '0b10', discarding the low 2 bits.
+        hex((u32_2 >>> 18) | 0x8000, 4)
+    }-${
+        // Use the 2 bits that were discarded above and instead discard weak low bits.
+        hex((u32_2 << 14) >>> 16, 4)
+    }${
+        hex(u32_3, 8)
+    }`;
+}
 
 /**
- * Returns a random-js `Random` instance, optionally seeded to make it deterministic.
- * @param seed - Optional seed to make the randomness source deterministic.
- * If not provided, this leverages Math.random() as an engine which is *not* deterministic.
+ * Construct a new IRandom instance, optionally seeding it with up to four 32b integers.
+ * If no seeds are provided, the PRNG is non-deterministically seeded using Math.random().
  */
-export function makeRandom(seed?: number | number[]): Random {
-    if (seed === undefined) {
-        return new Random();
-    }
-
-    const engine = Random.engines.mt19937();
-    if (typeof seed === "number") {
-        engine.seed(seed);
-    } else {
-        engine.seedWithArray(seed);
-    }
-
-    return new Random(engine);
+export function makeRandom(
+    ...seed: [] | [number] | [number, number] | [number, number, number] | [number, number, number, number]
+): IRandom {
+    const engine = new XSadd(...seed);
+
+    // RATIONALE: These methods are already bound.  (Technically, XSadd is  constructed to avoid use
+    //            of 'this' for a minor perf win, but the end result is the same.)
+
+    /* eslint-disable @typescript-eslint/unbound-method */
+    const real = (min: number, max: number) => distribution.real(engine.float64, min, max)();
+    const integer = (min: number, max: number) => distribution.integer(engine.uint53, min, max)();
+    const normal = (mu: number, sigma: number) => distribution.normal(engine.float64, mu, sigma)();
+    /* eslint-enable @typescript-eslint/unbound-method */
+
+    return {
+        bool: (probability = 0.5) => engine.float64() < probability,
+        integer,
+        normal,
+        pick: <T>(items: T[]) => items[integer(0, items.length - 1)],
+        real,
+        shuffle: <T>(items: T[]) => {
+            // Fisher–Yates shuffle
+            // See: https://en.wikipedia.org/wiki/Fisher%E2%80%93Yates_shuffle#The_modern_algorithm
+            for (let i = items.length - 1; i > 0; i--) {
+                const j = integer(0, i);
+                const tmp = items[i];
+                items[i] = items[j];
+                items[j] = tmp;
+            }
+        },
+        string: (length: number, alphabet = "123456789abcdefghijkmnopqrstuvwxyzABCDEFGHJKLMNPQRSTUVWXYZ") => {
```

Comment:
Note: The 'random-js' default alphabet is base64-ish [a-zA-Z9-0] plus - and _.

This is 'base58' which omits -, _, and [0OIl] for legibility.

Reply:
Yeah, seems fine. No usages I'm aware of cared much.